src/librustc_ast/util/comments.rs

```diff
@@ -216,8 +176,8 @@ pub fn gather_comments(sm: &SourceMap, path: FileName, src: String) -> Vec<Comme
                     }
                 }
             }
-            rustc_lexer::TokenKind::BlockComment { terminated } => {
-                if block_doc_comment_style(token_text, terminated).is_none() {
+            rustc_lexer::TokenKind::BlockComment { kind, terminated: _ } => {
+                if kind == rustc_lexer::CommentKind::Plain {
```

Comment:
Nit: could you `use rustc_lexer::CommentKind` instead?
Long paths like this are not normally used in rustc.

Reply:
Ack. I'll leave `rustc_lexer::TokenKind` paths in though, as there are two `TokenKind`s (and this is unrelated to PR)

Reply:
Hm, actually, let me take this back. In the `lexer/mod.rs`, if I do replacement, I get

```
    fn cook_doc_comment(
        &self,
        content_start: BytePos,
        content: &str,
        comment_kind: CommentKind,
        doc_kind: DocCommentKind,
    ) -> TokenKind {
```

The problem here is that `CommentKind` and `DocCommentKind` are from different crates, which are confusing.

Reply:
With this PR `CommentKind` from `rustc_ast` can be eliminated.

Reply:
In case of duplicated types `rust_lexer` version have to be qualified, yes.

Reply:
Hm, either I disagree, or I don't understand. The two `CommentKind`s are orthogonal axis: one is Inner/Outer, the other is Line/Block. ast's CommentKind (Line/Block) fits the current shape of the ast, it doesn't seem worth it to change it.

What can be merged is `rustc_lexer::CommentKind` and `rustc_ast::AttrStyle`, as they are both just `InnerOrOuter`. I think it's nicer than they are separate though, as they exist on different levels of abstraction, and `AttrStyle` is also serializable.

Reply:
Edited https://github.com/rust-lang/rust/pull/75642#issuecomment-676058846 to better show type correspondence. *Independently*, the naming is good :-(

Reply:
Ok, I didn't have the code in front of my eyes and mixed everything up.